src/library_dylink.js

```diff
@@ -20,12 +122,38 @@ var LibraryDylink = {
         value += memoryBase;
       }
       relocated[e] = value;
-      if (moduleLocal) {
-        moduleLocal[e] = value;
-      }
     }
+    updateGOT(relocated);
     return relocated;
   },
+
+  $reportUndefinedSymbols__deps: ['$GOT', '$resolveModuleSymbol'],
+  $reportUndefinedSymbols: function() {
+#if DYLINK_DEBUG
+    err('reportUndefinedSymbols');
+#endif
+    for (var symName in GOT) {
+      if (GOT[symName].value == 0) {
+        var value = resolveModuleSymbol(symName, true)
+#if DYLINK_DEBUG
+        err('assigning dynamic symbol from main module: ' + symName + ' -> ' + value);
+#endif
+        if (typeof value === 'function') {
+          GOT[symName].value = addFunctionWasm(value, value.sig);
```

Comment:
this uses `value.sig`, but is that not only defined for JS library functions? Where is it defined for other things?

Reply:
I'm not sure how a non-library JS function could be returned from resolveModuleSymbol.   Perhaps if a user somehow injects an extra symbol onto the module from the outside?

In that case it would behave just a like a library function that did not have `__sig` assigned at the second argument here would be undefined which would lead to a runtime failure adding it to the table and you would see `Missing signature argument to addFunction: foo`.

Reply:
I see, sounds good.